KsApi/models/graphql/adapters/Project+FetchProjectRewardsByIdQueryData.swift

```diff
@@ -13,9 +13,56 @@ extension Project {
 
   static func projectRewards(from data: GraphAPI.FetchProjectRewardsByIdQuery.Data) -> [Reward] {
     let projectRewards = data.project?.rewards?.nodes?
-      .compactMap { $0?.fragments.rewardFragment }
-      .compactMap { Reward.reward(from: $0) } ?? []
+      .compactMap { node -> (GraphAPI.RewardFragment, [ShippingRule]?)? in
+        guard let rewardFragment = node?.fragments.rewardFragment else { return nil }
 
-    return projectRewards
+        // These shipping rules are constructed from simplified versions of the shipping rules.
+        // Names are not localized and should not be shown to users, but they contain the data we
+        // need to calculate shipping, just in a different initial format.
+        let expandedShippingRules = node?.simpleShippingRulesExpanded?
+          .compactMap { node -> ShippingRule? in
+            guard let node,
+                  let idString = node.locationId, let locationId = decompose(id: idString)
+            else {
+              return nil
+            }
+
+            let name = node.locationName ?? ""
+            let location = Location.init(
+              country: node.country,
+              displayableName: name,
+              id: locationId,
+              localizedName: name,
+              name: name
+            )
+
+            let cost = node.cost.flatMap(Double.init) ?? 0.0
+            let estimatedMin = Money(
+              amount: node.estimatedMin.flatMap(Double.init) ?? 0.0,
+              currency: Money.CurrencyCode(rawValue: node.currency ?? "")
+            )
+            let estimatedMax = Money.init(
+              amount: node.estimatedMax.flatMap(Double.init) ?? 0.0,
+              currency: Money.CurrencyCode(rawValue: node.currency ?? "")
+            )
```

Comment:
Is it correct that these money values should show as 0 if they don't parse?

Reply:
We do this in some places already, I think, but you're right; I don't think it makes sense. I've updated this to require cost and the name as well, just in case we decide to use that later. I'm still following the same pattern that we currently have for this fetch, where something just returns nil (and is missing from the results) if it's malformed, instead of throwing an error. (estimatedMin and max will be `nil` instead, if they're missing or malformed)